src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2096,8 +2096,10 @@ private IOperation VisitConditionalExpression(IOperation condition, bool sense,
         {
             Debug.Assert(ITypeSymbolHelpers.IsBooleanType(condition.Type));
 
-            // PROTOTYPE(dataflow): Unwrap parenthesized and look for other places where it should happen
-            // PROTOTYPE(dataflow): Do not erase UnaryOperatorKind.Not if ProduceIsSense below will have to add it back.
+            while (condition.Kind == OperationKind.Parenthesized)
```

Comment:
>while (condition.Kind == OperationKind.Parenthesized) [](start = 12, length = 53)

It feels like this should be done inside the loop too, after each time we dig through a logical Not. Consider changing the loop below to:

while(true)
{
    Dig through parentesized
    Break if not logical not
    Dig through logical Not
}

Reply:
Please add relevant test.

---
In reply to: [193943036](https://github.com/dotnet/roslyn/pull/27611#discussion_r193943036) [](ancestors = 193943036)